File: JavaServer/Qipai4Core/src/net/silverfoxserver/core/socket/XmlInstruction.java
/*
 * SilverFoxServer: massive multiplayer game server for Flash, ...
 * VERSION:3.0
 * PUBLISH DATE:2015-9-2 
 * GITHUB:github.com/wdmir/521266750_qq_com
 * UPDATES AND DOCUMENTATION AT: http://www.silverfoxserver.net
 * COPYRIGHT 2009-2015 SilverFoxServer.NET. All rights reserved.
 * MAIL:521266750@qq.com
 */
package net.silverfoxserver.core.socket;

import java.io.UnsupportedEncodingException;
import org.jboss.netty.util.CharsetUtil;

/**
 *
 * @author ACER-FX
 */
public class XmlInstruction {
    
    /** 
	 封包
	 
	 @return 
	*/
//C# TO JAVA CONVERTER WARNING: Unsigned integer types have no direct equivalent in Java:
//ORIGINAL LINE: public override byte[] fengBao(string action,string content)
	//@Override
	public static byte[] fengBao(String action, String content) throws UnsupportedEncodingException
	{
		//最后加\0
		String s = "<msg t='sys'><body action='" + action + "'>" + content + "</body></msg>\0";

		//return Encoding.UTF8.GetBytes(resXml);
                return s.getBytes(CharsetUtil.UTF_8);//"UTF-8");
	}
        
        public static byte[] fengBao(String action) throws UnsupportedEncodingException
	{
		//最后加\0
		String s = "<msg t='sys'><body action='" + action + "'>" + "" + "</body></msg>\0";

		//return Encoding.UTF8.GetBytes(resXml);
                return s.getBytes(CharsetUtil.UTF_8);//"UTF-8");
	}

	/** 
	 封包ByDB
	 
	 @return 
	*/
//C# TO JAVA CONVERTER WARNING: Unsigned integer types have no direct equivalent in Java:
//ORIGINAL LINE: public override byte[] DBfengBao(string action, string content)
	//@Override
	public static byte[] DBfengBao(String action, String content) throws UnsupportedEncodingException
	{
		//最后加\0
		//DBS = data base system
		String s = "<msg t='DBS'><body action='" + action + "'>" + content + "</body></msg>\0";
                
		//return Encoding.UTF8.GetBytes(resXml);
                return s.getBytes("UTF-8");
	}

    
}

File: CSharpServer/Qipai4Core/Qipai4Core/net/silverfoxserver/core/socket/XmlInstruction.cs
/*
* SilverFoxServer: massive multiplayer game server for Flash, ...
* VERSION:3.0
* PUBLISH DATE:2015-9-2 
* GITHUB:github.com/wdmir/521266750_qq_com
* UPDATES AND DOCUMENTATION AT: http://www.silverfoxserver.net
* COPYRIGHT 2009-2015 SilverFoxServer.NET. All rights reserved.
* MAIL:521266750@qq.com
*/
using System;
using System.Collections.Generic;
using System.Text;
//
using System.Runtime.Serialization;
using System.Runtime.Serialization.Formatters.Binary;
//
using net.silverfoxserver.core.service;

namespace net.silverfoxserver.core.socket
{
    /// <summary>
    /// Xml指令构建器
    /// </summary>
    public class XmlInstruction
    {
        
        /// <summary>
        /// 封包
        /// </summary>
        /// <returns></returns>
        public static byte[] fengBao(string action, string content)
        {
            //最后加\0
            string resXml = "<msg t='sys'><body action='" + action + "'>" +
                                            content + "</body></msg>\0";

            return Encoding.UTF8.GetBytes(resXml);
        }

        /// <summary>
        /// 封包ByDB
        /// </summary>
        /// <returns></returns>
        public static byte[] DBfengBao(string action, string content)
        {
            //最后加\0
            //DBS = data base system
            string resXml = "<msg t='DBS'><body action='" + action + "'>" +
                                        content + "</body></msg>\0";

            return Encoding.UTF8.GetBytes(resXml);
        }

    }
}
